plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -92,26 +95,44 @@
 	public static final String SERVER_START_ON_CREATION = "org.jboss.tools.openshift.SERVER_START_ON_CREATION";
 
 	private static final Collection<String> EAP_LIKE_KEYWORDS = Collections.unmodifiableCollection(Arrays.asList("eap", "wildfly"));
+	
+	private static final Collection<String> SERVER_ADDAPTER_ALLOWED_RESOURCE_TYPESS = Collections.unmodifiableCollection(Arrays.asList(ResourceKind.ROUTE,
+	                                                                                                                                   ResourceKind.SERVICE,
+	                                                                                                                                   ResourceKind.REPLICATION_CONTROLLER,
+	                                                                                                                                   ResourceKind.DEPLOYMENT_CONFIG,
+	                                                                                                                                   ResourceKind.POD));
+	
+	/**
+	 * Checks if the resource is allowed for OpenShift server adapter.
+	 * 
+	 * @param resource the OpenShift resource
+	 * @return true if allowed
+	 */
+	public static boolean isAllowedForServerAdapter(IResource resource) {
+	    return SERVER_ADDAPTER_ALLOWED_RESOURCE_TYPESS.stream().anyMatch(kind -> kind.equals(resource.getKind()));
```

Comment:
`return SERVER_ADDAPTER_ALLOWED_RESOURCE_TYPESS.contains(resource.getKind());` should work fine